docs/classes.rst

```diff
@@ -1012,4 +1012,87 @@ custom ``__init__``-style constructors.
+     public:
+       static std::unique_ptr<Pet> make(std::string name);
+       void speak();
+   };
+
+You can use :cpp:func:`.def_static() <class_::def_static>` to
+produce bindings that let you write ``Pet.make("Fido")`` in Python,
+just like you would write ``Pet::make("Fido")`` in C++. But sometimes it's
+nice to provide a more Pythonic interface than that, like ``Pet("Fido")``.
+To do that, nanobind lets you override ``__new__``.
+
+Since this is a rarely-used feature in Python, let's recap. Object
+initialization in Python occurs in two phases:
+
+* the *constructor*, ``__new__``, allocates memory for the object;
+
+* the *initializer*, ``__init__``, sets up the object's initial state.
+
+So far, all the ways we've seen of binding C++ constructors
+(:cpp:struct:`nb::init\<..\>() <init>`, ``.def("__init__", ...)``)
+produce Python object *initializers*. nanobind augments these with its own
+Python object constructor, which allocates a Python object that has
+space in its memory layout for the C++ object to slot in.
+The ``__init__`` method then fills in that space by calling a C++ constructor.
+
+This split between ``__new__`` and ``__init__`` has a lot of benefits,
+including a reduction in unnecessary allocations, but it does mean that
+anything created from Python must be able to control where its
+C++ innards are stored. Sometimes, as with the example of ``Pet`` above,
+that's not feasible. In such cases, you can duck down a layer and override
+``__new__`` directly:
+
+.. code-block:: cpp
+
+   nb::class_<Pet>(m, "Pet")
+       .def(nb::new_(&Pet::make))
+       .def("speak", &Pet::speak);
+
+It also works fine to pass a lambda as the argument of
+:cpp:struct:`nb::new_ <new_>`. This is most useful when the lambda
+returns a pointer or smart pointer; if it's returning a value, then
+``.def("__init__", ...)`` will have better performance.
+
+Passing :cpp:struct:`nb::new_ <new_>` to :cpp:func:`.def() <class_::def>`
+here creates two magic methods on ``Pet``:
+
+* A ``__new__`` that uses the given function to produce a new ``Pet``.
+  It is converted to a Python object in the same way as the return value
+  of any other function you might write bindings for. In particular,
+  you can pass a :cpp:enum:`nb::rv_policy <rv_policy>` as an additional
+  argument to :cpp:func:`.def() <class_::def>` to control how this conversion
+  occurs.
+
+* A ``__init__`` that takes the same arguments as ``__new__`` but
+  performs no operation.  This is necessary because Python automatically
+  calls ``__init__`` on the object returned by ``__new__`` in most cases.
+
+If you need even more control, such as because you need to access the
+type object that Python passes as the first argument of ``__new__``
+(which :cpp:struct:`nb::new_ <new_>` discards), you can write a
+``.def_static("__new__", ...)`` and matching ``.def("__init__", ...)``
+yourself.
+
+.. note::
+   Even if you provide a custom ``__new__`` that takes additional arguments,
```

Comment:
I think there needs to be a black line between the RST directive and the body.

Reply:
It looks fine locally. I don't think there even needs to be a newline; I've written `.. note:: words words words` in other docs before.